tools/setup.py

```diff
@@ -56,33 +56,38 @@ def prepare_executable(original_nso: Optional[Path]):
     if not TARGET_ELF_PATH.is_file():
         setup.fail("internal error while preparing executable (missing ELF); please report")
 
-def create_build_dir(ver):
+def get_build_dir():
+    return setup.ROOT / "build"
+
+def create_build_dir(ver, cmake_backend):
     if(ver != Version.VER_100): return # TODO remove this when multiple versions should be built
-    build_dir = setup.ROOT / "build" # ("build-"+ver.value)
+    build_dir = get_build_dir()
     if build_dir.is_dir():
         print(">>> build directory already exists: nothing to do")
         return
 
     subprocess.check_call(
-        ("cmake -GNinja -DCMAKE_CXX_FLAGS=-D"+ver.name+" -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_TOOLCHAIN_FILE=toolchain/ToolchainNX64.cmake -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -B "+str(build_dir)).split(" "))
+        ['cmake', '-G', cmake_backend, f'-DCMAKE_CXX_FLAGS=-D"{ver.name}"', '-DCMAKE_BUILD_TYPE=RelWithDebInfo', '-DCMAKE_TOOLCHAIN_FILE=toolchain/ToolchainNX64.cmake', '-DCMAKE_CXX_COMPILER_LAUNCHER=ccache', '-B', str(build_dir)])
```

Comment:
```suggestion
        ['cmake', '-G', cmake_backend, f'-DCMAKE_CXX_FLAGS=-D{ver.name}', '-DCMAKE_BUILD_TYPE=RelWithDebInfo', '-DCMAKE_TOOLCHAIN_FILE=toolchain/ToolchainNX64.cmake', '-DCMAKE_CXX_COMPILER_LAUNCHER=ccache', '-B', str(build_dir)])
```
Not entirely sure on that one, but I think with the current state it generates `"` there - which is not correct.

Reply:
I didn't notice those quotes staying there after i switched the args from " to '. oops
pushed a fix